modules/scientiamobile/wurfl_devicedetection/config_test.go

```diff
@@ -0,0 +1,121 @@
+package wurfl_devicedetection
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewConfig(t *testing.T) {
+	tests := []struct {
+		name        string
+		input       json.RawMessage
+		expectedErr bool
+		validate    func(t *testing.T, cfg config)
+	}{
+		{
+			name: "Valid config with default cache size",
+			input: json.RawMessage(`{
+				"wurfl_snapshot_url": "http://example.com/wurfl-data",
+				"wurfl_file_path": "/tmp/wurfl.zip"
+			}`),
+			expectedErr: false,
+			validate: func(t *testing.T, cfg config) {
+				assert.Equal(t, "http://example.com/wurfl-data", cfg.WURFLSnapshotURL)
+				assert.Equal(t, "/tmp/wurfl.zip", cfg.WURFLFilePath)
+				assert.Equal(t, defaultCacheSize, cfg.WURFLEngineCacheSize())
+			},
+		},
+		{
+			name: "Valid config with custom cache size",
+			input: json.RawMessage(`{
+				"wurfl_snapshot_url": "http://example.com/wurfl-data",
+				"wurfl_file_path": "/tmp/wurfl.zip",
+				"wurfl_cache_size": 5000,
+				"wurfl_run_updater": true
+			}`),
+			expectedErr: false,
+			validate: func(t *testing.T, cfg config) {
+				assert.Equal(t, "5000", cfg.WURFLEngineCacheSize())
+			},
+		},
+		{
+			name: "Invalid config - missing wurfl_snapshot_url",
```

Comment:
`name` says "Invalid" but `expectedErr` is false. Shouldn't this be "Valid" instead?